engines/wintermute/base/base_game.cpp

```diff
@@ -1929,6 +1986,95 @@ bool BaseGame::scCallMethod(ScScript *script, ScStack *stack, ScStack *thisStack
 		return STATUS_OK;
 	}
 
+#ifdef ENABLE_FOXTAIL
+	//////////////////////////////////////////////////////////////////////////
+	// [FoxTail] GetScreenType
+	// Returns 0 on fullscreen and 1 on window
+	// Used to init and update controls at options.script and methods.script
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "GetScreenType") == 0) {
+		stack->correctParams(0);
+		int type = !g_system->getFeatureState(OSystem::kFeatureFullscreenMode);
+		stack->pushInt(type);
+
+		return STATUS_OK;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// [FoxTail] GetScreenMode
+	// Returns integer to be used as a pixelization mode multiplier
+	// Used to init and update controls at options.script and methods.script
+	// This implementation fakes window size as 2*320 x 2*180
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "GetScreenMode") == 0) {
+		stack->correctParams(0);
+		stack->pushInt(2);
```

Comment:
Those "screen modes" are basically duplicate of ScummVM's functionality of "2x", "3x" renderers. Game starts with "2x" mode by default (displayed as "640x360" button) and can be switched to "3x","4x","5x", etc, given enough screen space on actual display. 

![изображение](https://user-images.githubusercontent.com/5786428/71771961-26a00100-2f54-11ea-94a5-dd8c6da2e61b.png)
![изображение](https://user-images.githubusercontent.com/5786428/71771964-2dc70f00-2f54-11ea-848a-a89d42588588.png)

We have discussed this before with @tobiatesan and @somaen, and decided to disable ingame controls of render modes, because ScummVM supports much more then simple "2x","3x" - there are different 2x filters like "2x", "HQ2x", "2xSAI", "AdvMAME2x", "TV2x". There is no way to pass list of those modes to game script, also game code does not expect "1x" mode to exist, so the best thing we could implement is switching between 2 modes - "2x" and "3x", which is much less than ScummVM is capable of.

It's safe and wise to force the game to always show "640x360" in-game button that does nothing (because game script think that hardware display size is 641x361) and let user control render modes with Alt+Ctrl+N combinations and other common ways shared for all games.

![изображение](https://user-images.githubusercontent.com/5786428/71771978-636bf800-2f54-11ea-9fe3-9a991a765c83.png)

Reply:
Thanks for the explanation, I agree that we can force the game to always show a fixed value for the resolution, since we already have this functionality, to a much better degree, in ScummVM